app/controllers/scholars_archive/handles_controller.rb

```diff
@@ -0,0 +1,97 @@
+module ScholarsArchive
+  class HandlesController < ApplicationController
+    before_filter :verify_handle_id, only: [:handle_show, :handle_download]
+    skip_before_action :check_d2h_http_header_auth
+
+    def handle_file_404
+      @handle_uri = "#{params[:handle_id]}/#{params[:id]}"
+      @file = "#{params[:file]}"
+      work = find_work
+      @work_path = "/concern/#{work.class.to_s.downcase.pluralize}/#{work.id}"
+    end
+
+    def handle_work_404
+
+    end
+
+    def handle_show
+      work = find_work
+      if work.nil?
+        ScholarsArchive::HandleErrorLoggingService.log_no_work_found_error(params)
+        redirect_to "/404/work_not_found"
+      else
+        redirect_to "/concern/#{work.class.to_s.downcase.pluralize}/#{work.id}"
+      end
+    end
+
+    def handle_download
+      work = find_work
+      filesets = filesets_for_work(work) unless work.nil?
+      if work.nil?
+        ScholarsArchive::HandleErrorLoggingService.log_no_work_found_error(params)
+        redirect_to "/404/work_not_found"
+      elsif filesets.empty?
+        ScholarsArchive::HandleErrorLoggingService.log_no_files_found_error(params, work, construct_handle_url(params[:handle_id], params[:id]))
+        redirect_to "/404/file_not_found/#{params[:handle_id]}/#{params[:id]}/#{params[:file]}"
+      elsif filesets.length > 1
+        ScholarsArchive::HandleErrorLoggingService.log_too_many_files_found_error(params, work, construct_handle_url(params[:handle_id], params[:id]))
+        redirect_to "/concern/#{work.class.to_s.downcase.pluralize}/#{work.id}"
+      else
+        redirect_to "/downloads/#{filesets.first.id}"
+      end
+    end
+
+    private
+      def verify_handle_id
```

Comment:
To me `:handle_id` means the whole thing, and using it only for the prefix is confusing. From the Handle docs: `Handles consist of a prefix which identifies a "naming authority" and a suffix which gives the "local name" of a resource.`.
I would suggest changing all instances (including methods and log output):
* change `:handle_id` to `:handle_prefix`
* change `:id` to `:handle_localname`